contribs/gnodepgraph/depgraph.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+type depGraphCfg struct {
+	verbose        bool
+	rootDir        string
+	output         string
+	multipleGraphs bool
+}
+
+func newDepGraphCmd(io commands.IO) *commands.Command {
+	cfg := &depGraphCfg{}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "gnodepgraph",
+			ShortUsage: "gnodepgraph [flags] <package> [<package>...]",
+			ShortHelp:  "generates dependency graphs for the specified packages",
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execDepGraph(cfg, args, io)
+		},
+	)
+}
+
+func (c *depGraphCfg) RegisterFlags(fs *flag.FlagSet) {
+	rootDir := gnoenv.RootDir()
+	fs.BoolVar(&c.verbose, "v", false, "verbose output when lintning")
+	fs.StringVar(&c.rootDir, "root-dir", rootDir, "clone location of github.com/gnolang/gno (gno tries to guess it)")
+	fs.StringVar(&c.output, "o", "depgraph", "output (file if single graph, dir if multiple graphs)")
+	fs.BoolVar(&c.multipleGraphs, "m", false, "make a separate graph for each package")
+}
+
+func execDepGraph(cfg *depGraphCfg, args []string, io commands.IO) error {
+	if len(args) < 1 {
+		return flag.ErrHelp
+	}
+
+	var (
+		verbose        = cfg.verbose
+		rootDir        = cfg.rootDir
+		output         = cfg.output
+		multipleGraphs = cfg.multipleGraphs
+		allPkgs        gnomod.PkgList
+	)
+
+	// first we walk through all the folders and collect all packages
+	for _, arg := range args {
+		pkgs, err := gnomod.ListPkgs(arg)
+		if err != nil {
+			return fmt.Errorf("error in parsing gno.mod: %w", err)
+		}
+
+		allPkgs = append(allPkgs, pkgs...)
+	}
+
+	// make one big graph (eg. for the entire examples/ dir)
+	if !multipleGraphs {
+		nodeData := ""  // nodes
+		graphData := "" // edges
```

Comment:
nit: Using a `strings.Builder` or `bytes.Buffer` along with `fmt.Fprintf` and passing an `io.Writer` to `buildGraphData` would have been a cleaner approach than directly writing the output to a string directly.